From 1a563bb767075ce5623f3fe4eb8c4654528eeea6 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Mon, 27 Jul 2026 15:04:32 +0000
Subject: [PATCH 57/64] wifi: mt76: mt7915: update SKU power limits after
 changing antennas

Changing the antenna configuration updates the stream capabilities but
left the per-path SKU power limits and path delta compensation stale
until the next channel switch. Reapply the SKU table like mt7996
already does.

Link: https://patch.msgid.link/20260727150434.1778520-13-nbd@nbd.name
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit 8baa3813f494949cc22f75edd8d6dc4664fe4ddb)
---
 mt7915/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7915/main.c b/mt7915/main.c
index 075106ec..a414107c 100644
--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -1162,6 +1162,7 @@ mt7915_set_antenna(struct ieee80211_hw *hw, int radio_idx, u32 tx_ant, u32 rx_an
 	mt76_set_stream_caps(phy->mt76, true);
 	mt7915_set_stream_vht_txbf_caps(phy);
 	mt7915_set_stream_he_caps(phy);
+	mt7915_mcu_set_txpower_sku(phy);
 
 	mutex_unlock(&dev->mt76.mutex);
 
